packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -194,10 +170,36 @@ export abstract class TelemetryLogger implements ITelemetryLogger {
         this.send(perfEvent);
     }
 
-    protected prepareEvent(event: ITelemetryBaseEvent): ITelemetryBaseEvent {
+    protected redactTaggedUserData(event: ITelemetryBaseEvent, propKey: string) {
+        const prop = event[propKey];
+        const { value, tag } = isTaggedTelemetryPropertyValue(prop)
+            ? prop
+            : { value: prop, tag: undefined };
+
+        switch (tag) {
+            case undefined:
+                // No tag means we can log plainly
+            case TelemetryDataTag.PackageData:
+                // For Microsoft applications, PackageData is safe for now
+                // (we don't load 3P code in 1P apps)
+                // But this determination really belongs in the host layer
+                event[propKey] = value;
+            case TelemetryDataTag.UserData:
+                // Strip out anything tagged explicitly as PII.
+                // Alternate strategy would be to hash these props
+                // console.log("userdata");
+                event[propKey] = "REDACTED (UserData)";
+            default:
+                // If we encounter a tag we don't recognize
+                // then we must assume we should scrub.
+                event[propKey] = "REDACTED (unknown tag)";
+        }
+    }
+
+    protected prepareEvent(event: ITelemetryBaseEvent, eventCanSupportTags?: boolean): ITelemetryBaseEvent {
```

Comment:
Looks like you haven't updated the call sites for prepareEvent to pass in the right value for `eventCanSupportTags`.  Probably should be required, btw.

Reply:
Will do, stopped partway after seeing Vlad's comments